packages/@internationalized/number/test/NumberParser.test.js

```diff
@@ -137,10 +142,82 @@ describe('NumberParser', function () {
+          {requiredKeys: ['style', 'unit']}
+        )
+      );
+      const genericOptsArb = fc.record({
+        localeMatcher: fc.constantFrom('best fit', 'lookup'),
+        unitDisplay: fc.constantFrom('narrow', 'short', 'long'),
+        useGrouping: fc.boolean(),
+        minimumIntegerDigits: fc.integer({min: 1, max: 21}),
+        minimumFractionDigits: fc.integer({min: 0, max: 20}),
+        maximumFractionDigits: fc.integer({min: 0, max: 20}),
+        minimumSignificantDigits: fc.integer({min: 1, max: 21}),
+        maximumSignificantDigits: fc.integer({min: 1, max: 21})
+      }, {requiredKeys: []});
+
+      // We restricted the set of possible values to avoid unwanted overflows to infinity and underflows to zero
+      // and stay in the domain of legit values.
+      const DOUBLE_MIN = Number.EPSILON;
+      const valueArb = fc.tuple(
+        fc.constantFrom(1, -1),
+        fc.double({next: true, noNaN: true, min: DOUBLE_MIN, max: 1 / DOUBLE_MIN})
+      ).map(([sign, value]) => sign * value);
+
+      it('should fully reverse NumberFormat', function () {
+        fc.assert(
+          fc.property(
+            valueArb, localesArb, styleOptsArb, genericOptsArb,
+            function (d, locales, styleOpts, genericOpts) {
+              const opts = {...styleOpts, ...genericOpts};
+              // is there some way to combine this with the above logic?
+              if (styleOpts.style === 'percent') {
+                opts.minimumFractionDigits = genericOpts.minimumFractionDigits > 18 ? 18 : genericOpts.minimumFractionDigits;
+                opts.maximumFractionDigits = genericOpts.maximumFractionDigits > 18 ? 18 : genericOpts.maximumFractionDigits;
+              }
+              fc.pre(opts.minimumFractionDigits === undefined || opts.maximumFractionDigits === undefined || opts.minimumFractionDigits <= opts.maximumFractionDigits);
+              fc.pre(opts.minimumSignificantDigits === undefined || opts.maximumSignificantDigits === undefined || opts.minimumSignificantDigits <= opts.maximumSignificantDigits);
+              // invalid currency maximumSignificantDigits, getting an error about too many pre-conditions
+              // fc.pre(!(opts.style === 'currency' && opts.maximumSignificantDigits <= 2));
+              // this set of options for da-DK results in nothing but the unit being displayed, it seems to be a bug in the browser? (new Intl.NumberFormat('da-DK', {"style":"unit","unit":"kilometer-per-hour", 'unitDisplay': 'long', "maximumSignificantDigits":5})).format(0.2)
+              // fc.pre(!(opts.style === 'unit' && opts.unit === 'kilometer-per-hour' && opts.unitDisplay === 'long' && (Math.abs(d) < 1 && Math.abs(d) > 0) && locales === 'da-DK'));
+              const formatter = new Intl.NumberFormat(locales, opts);
+              const parser = new NumberParser(locales, opts);
+
+              // is there some way to combine this with the above logic?
```

Comment:
Yes you can!

Basically you could do something like (if I understand well the need):
```js
const inputsArb = fc.tuple(valueArb, localesArb, styleOptsArb, genericOptsArb)
  .map(([d, locales, styleOpts, genericOpts]) => {
    const opts = {...styleOpts, ...genericOpts};
    if (styleOpts.style === 'percent') {
      opts.minimumFractionDigits = genericOpts.minimumFractionDigits > 18 ? 18 : genericOpts.minimumFractionDigits;
      opts.maximumFractionDigits = genericOpts.maximumFractionDigits > 18 ? 18 : genericOpts.maximumFractionDigits;
     }
     return {d,opts,locales};
  })
  .filter(({opts}) => opts.minimumFractionDigits === undefined || opts.maximumFractionDigits === undefined || opts.minimumFractionDigits <= opts.maximumFractionDigits)
  .filter(({opts}) => opts.minimumSignificantDigits === undefined || opts.maximumSignificantDigits === undefined || opts.minimumSignificantDigits <= opts.maximumSignificantDigits)
  .map(({d,opts,locales}) => {
    let adjustedNumberForFractions = d;
    if (Math.abs(d) < 1 && opts.minimumFractionDigits && opts.minimumFractionDigits > 1) {
      adjustedNumberForFractions = d * (10 ** (opts.minimumFractionDigits || 2));
    } else if (Math.abs(d) > 1 && opts.minimumFractionDigits && opts.minimumFractionDigits > 1) {
      adjustedNumberForFractions = d / (10 ** (opts.minimumFractionDigits || 2));
    }
    return {adjustedNumberForFractions,opts,locales}
  });

fc.assert(fc.property(
  inputsArb,
  function ({adjustedNumberForFractions, locales, opts}) {
    // no more need for fc.pre or computation of adjustedNumberForFractions
```